File: runtime/go/mica/method.go
package mica

import "google.golang.org/protobuf/proto"

type RPCMethod struct {
	Service     string
	Method      string
	NewRequest  func() proto.Message
	NewResponse func() proto.Message
}

func (m RPCMethod) ContractID() string {
	return m.Service + "." + m.Method
}

File: runtime/python/src/mica/method.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True)
class RpcMethod:
    service: str
    method: str
    request_type: type
    response_type: type

    @property
    def contract_id(self) -> str:
        return f"{self.service}.{self.method}"
